arangod/RocksDBEngine/RocksDBCollection.cpp

```diff
@@ -1301,6 +1301,21 @@ arangodb::Result RocksDBCollection::fillIndexes(transaction::Methods* trx,
   return Result();
 }
 
+namespace {
+template<typename F>
+void reverseIdxOps(std::vector<std::shared_ptr<Index>> const& vector,
+                   std::vector<std::shared_ptr<Index>>::const_iterator& it,
+                   F&& op) {
+  while (it != vector.begin()) {
+    it--;
+    auto* rIdx = static_cast<RocksDBIndex*>(it->get());
+    if (rIdx->type() == Index::TRI_IDX_TYPE_IRESEARCH_LINK) {
+      std::forward<F>(op)(rIdx);
+    }
+  }
+}
+}
+
 Result RocksDBCollection::insertDocument(arangodb::transaction::Methods* trx,
```

Comment:
don't we need this for the remove operation too ?

Reply:
Added reversal for update and remove